modules/codegen/src/smithy4s/codegen/internals/SmithyToIR.scala

```diff
@@ -932,13 +930,69 @@ private[codegen] class SmithyToIR(model: Model, namespace: String) {
     case ConstraintTrait(tr) => Hint.Constraint(toTypeRef(tr), unfoldTrait(tr))
   }
 
+  private def streamingOperation(
+      op: OperationShape
+  ): (Option[Shape], Option[Shape]) = {
+    def forTarget(id: ShapeId): Option[MemberShape] = {
+      model.getShape(id).asScala.flatMap { shape =>
+        shape.members().asScala.find(isStreaming)
+      }
+    }
+    (
+      op.getInput().asScala.flatMap(forTarget),
+      op.getOutput().asScala.flatMap(forTarget)
+    )
+  }
+
   private def documentationHint(shape: Shape): Option[Hint] = {
     def split(s: String) =
       s.replace("*/", "\\*\\/").linesIterator.toList
     val shapeDocs = shape
       .getTrait(classOf[DocumentationTrait])
       .asScala
       .foldMap(doc => split(doc.getValue()))
+
+    val operationDocs: List[String] = {
+      shape
+        .asOperationShape()
+        .asScala
+        .toList
+        .flatMap { op =>
+          streamingOperation(op) match {
+            case (Some(in), Some(out)) =>
+              val maybeDoc = for {
+                inMem <- in
+                  .asMemberShape()
+                  .asScala
+                  .map(_.getMemberName)
+                outMem <- out
+                  .asMemberShape()
+                  .asScala
+                  .map(_.getMemberName)
+              } yield s"This operation uses @streaming on both the input (${inMem}) and the output (${outMem})"
+              maybeDoc.toList
+            case (Some(in), None) =>
+              val maybeDoc = for {
+                inMem <- in
+                  .asMemberShape()
+                  .asScala
+                  .map(_.getMemberName)
+              } yield s"This operation uses @streaming on the input (${inMem})."
+              maybeDoc.toList
+            case (None, Some(out)) =>
+              val maybeDoc = for {
+                outMem <- out
+                  .asMemberShape()
+                  .asScala
+                  .map(_.getMemberName)
+              } yield s"This operation uses @streaming on the output (${outMem})."
+              maybeDoc.toList
```

Comment:
**question:** shall we have some examples of this in `bootstrapped`?

Reply:
ugh actually I'm just blind/dumb and missed these examples. I don't think it's a must to have the one with both

Reply:
that said, it might be good to have such an example regardless of this documentation PR.